daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -553,15 +553,23 @@ cleanTests baseDir = testGroup "daml clean"
                                 ]
 
 templateTests :: TestTree
-templateTests = testGroup "templates"
+templateTests = testGroup "templates" $
     [ testCase name $ do
           withTempDir $ \dir -> withCurrentDirectory dir $ do
-              callCommandSilent $ unwords ["daml", "new", "foobar", name]
+              callCommandSilent $ unwords ["daml", "new", "foobar", "--template", name]
               withCurrentDirectory (dir </> "foobar") $ callCommandSilent "daml build"
     | name <- templateNames
+    ] <>
+    [ testCase "quickstart-java, positional template" $ do
+          -- Verify that the old syntax for `daml new` still works.
+          withTempDir $ \dir -> withCurrentDirectory dir $ do
+              callCommandSilent "daml new foobar quickstart-java"
+              contents <- readFileUTF8 "foobar/daml.yaml"
+              assertInfixOf "name: quickstart" contents
```

Comment:
Shouldn't the name of the project be `foobar`?

Reply:
No, `quickstart-java` is like the old `create-daml-app` in that it has a hardcoded name so this is a cheap way of making sure we pick up the right project.